backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapper.java

```diff
@@ -22,13 +25,27 @@
 @Mapper
 public interface SubscriptionModelMapper {
 
+	final EmbeddedWrappers embeddedWrappers = new EmbeddedWrappers(false);
+
 	@Nullable
 	default CollectionModel<SubscriptionModel> toModel(String userId, @Nullable Iterable<Subscription> subscriptions) {
 		final var subscriptionModels = StreamSupport.stream(subscriptions.spliterator(), false)
-			.map(subscription -> toModel(userId, subscription)).toList();
+				.map(subscription -> toModel(userId, subscription)).toList();
+
+		final var collection = CollectionModel.of(subscriptionModels)
+				.add(linkTo(methodOn(SubscriptionsController.class).getSubscriptionsByUserId(userId)).withSelfRel());
+
+		return wrapCollection(collection, SubscriptionModel.class);
+	}
 
-		return CollectionModel.of(subscriptionModels)
-			.add(linkTo(methodOn(SubscriptionsController.class).getSubscriptionsByUserId(userId)).withSelfRel());
+	@SuppressWarnings({ "unchecked" })
+	default <C> CollectionModel<C> wrapCollection(CollectionModel<C> collectionModel, Class<C> type) {
```

Comment:
Since this method is generic and can be applied to any collection, it should be moved to a super class.